File: cpp/program/Stasiun.cpp
#pragma once
#include <iostream>
#include <string>
using namespace std;

// Class Stasiun
class Stasiun {
// Attributes
private:
    string kodeStasiun;
    string namaStasiun;
    string tipeStasiun;

public:
    // Default Constructor
    Stasiun() {
        kodeStasiun = "";
        namaStasiun = "";
        tipeStasiun = "";
    }
    // Parameterized Constructor
    Stasiun(string kodeStasiun, string namaStasiun, string tipeStasiun) {
        this->kodeStasiun = kodeStasiun;
        this->namaStasiun = namaStasiun;
        this->tipeStasiun = tipeStasiun;
    }

    // Method to print Stasiun information
    void printInfo() {
        cout << "(" << kodeStasiun << ") " << namaStasiun << " - " << tipeStasiun << endl;
    }

    // Getter and Setter for kodeStasiun
    string getKodeStasiun() const { return this->kodeStasiun; }
    void setKodeStasiun(string kode) { this->kodeStasiun = kode; }

    // Getter and Setter for namaStasiun
    string getNamaStasiun() const { return this->namaStasiun; }
    void setNamaStasiun(string nama) { this->namaStasiun = nama; }

    // Getter and Setter for tipeStasiun
    string getTipeStasiun() const { return this->tipeStasiun; }
    void setTipeStasiun(string tipe) { this->tipeStasiun = tipe; }

    // Destructor
    ~Stasiun() {}
};

File: java/program/Stasiun.java
public class Stasiun{
    // Attributes
    private String kodeStasiun;
    private String namaStasiun;
    private String tipeStasiun;

    // Default Constructor
    public Stasiun() {
        this.kodeStasiun = "";
        this.namaStasiun = "";
        this.tipeStasiun = "";
    }

    // Parameterized Constructor    
    public Stasiun(String kodeStasiun, String namaStasiun, String tipeStasiun) {
        this.kodeStasiun = kodeStasiun;
        this.namaStasiun = namaStasiun;
        this.tipeStasiun = tipeStasiun;
    }

    // Method to print stasiun info
    public void printInfo() {
        System.out.println("(" + kodeStasiun + ") " + namaStasiun + " - " + tipeStasiun);
    }

    // Getter and Setter for kodeStasiun
    public String getKodeStasiun() {return this.kodeStasiun;}
    public void setKodeStasiun(String kode) {this.kodeStasiun = kode;}

    // Getter and Setter for namaStasiun
    public String getNamaStasiun() {return this.namaStasiun;}
    public void setNamaStasiun(String nama) {this.namaStasiun = nama;}

    // Getter and Setter for tipeStasiun
    public String getTipeStasiun() {return this.tipeStasiun;}
    public void setTipeStasiun(String tipe) {this.tipeStasiun = tipe;}
}